backend/internal/utils/validators.go

```diff
@@ -1,19 +1,19 @@
-// internal/utils/validators.go
-package utils
-
-import (
-	"regexp"
-)
-
-// Check if the string is a valid email address
-func IsEmail(input string) bool {
-	// Define the regular expression for validating an email address
-	var emailRegex = regexp.MustCompile(`^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$`)
-	return emailRegex.MatchString(input)
-}
-
-// Check if the string is a valid UUID
-func IsUUID(input string) bool {
-	uuidRegex := regexp.MustCompile(`^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$`)
-	return uuidRegex.MatchString(input)
-}
+// internal/utils/validators.go
+package utils
+
+import (
+	"regexp"
+)
+
+// Check if the string is a valid email address
+func IsEmail(input string) bool {
+	// Define the regular expression for validating an email address
+	var emailRegex = regexp.MustCompile(`^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$`)

```

Comment:
these regex are being compiled on each function call. they should be precompiled in a private var, then used in this function